src/Modal/Modal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {memo, forwardRef} from 'react'
+import {Modal, View, StyleSheet, Pressable, ViewStyle, StyleProp} from 'react-native'
+
+interface IBaseModalProps {
+  /**
+   * Defines the visible of the Modal
+   * default false
+   */
+  visible?: boolean
+  /**
+   * Defines the dismissable of the Modal
+   * default true
+   */
+  dismissable?: boolean
+  /**
+   * Children element
+   */
+  children: React.ReactNode
+  /**
+   * Defines the animationType
+   * default 'fade'
+   */
+  animationType?: 'none' | 'slide' | 'fade' | undefined
+  /**
+   * Defines background color of backdrop
+   * default 'rgba(0,0,0,0.5)'
+   */
+  backdropColor?: string
+  /**
+   * ViewStyle for content container view
+   */
+  contentContainerStyle?: StyleProp<ViewStyle>
+  /**
+   * onClose method for close the Modal
+   */
+  onClose?: () => void
+}
+
+const BaseModal = forwardRef<Modal, IBaseModalProps>(
+  (
+    {
+      visible = false,
+      dismissable = true,
+      animationType = 'fade',
+      backdropColor = 'rgba(0,0,0,0.5)',
+      contentContainerStyle,
+      onClose,
+      children,
+    },
+    ref,
+  ) => {
+    return (
+      <Modal
+        ref={ref}
+        visible={visible}
+        transparent
+        animationType={animationType}
+        onRequestClose={() => (dismissable ? onClose && onClose() : undefined)}>
+        <Pressable
+          style={StyleSheet.flatten([styles.backdrop, {backgroundColor: backdropColor}])}
```

Comment:
what is different between stylesheet compose and flatten?

Reply:
Compose: Combines two styles such that style2 will override any styles in style1. If either style is falsy, the other one is returned without allocating an array, saving allocations and maintaining reference equality for PureComponent checks.

Flatten: Flattens an array of style objects, into one aggregated style object.

Actually we will update according to our convention